src/mslice/plotting/plot_window/cut_plot.py

```diff
@@ -178,19 +178,16 @@ def update_legend(self, line_data=None):
         handles_to_show = []
         handles, labels = axes.get_legend_handles_labels()
         if line_data is None:
-            i = 0
-            for handle, label in zip(handles, labels):
+            for i, (handle, label) in enumerate(zip(handles, labels)):
                 if self.legend_visible(i):
                     labels_to_show.append(label)
                     handles_to_show.append(handle)
-                i += 1
         else:
-            containers = axes.containers
-            for i in range(len(containers)):
-                if line_data[i]['legend']:
-                    handles_to_show.append(handles[i])
-                    labels_to_show.append(line_data[i]['label'])
-                self._legends_visible[i] = line_data[i]['legend']
+            for line, handle in zip(line_data, handles):
+                if line['legend']:
+                    handles_to_show.append(handle)
+                    labels_to_show.append(line['label'])
+            self._legends_visible = [line['legend'] for line in line_data]
 
```

Comment:
The `axes.containers `variable was just being used to loop through `handles` and` line_data`, so I removed it and used `zip` instead. Actually, inside this `else` statement, `handles` **will be the same as** `axes.containers` (I checked this in the source code of matplotlib), so it is best removed anyway.